notebooks/11_temporal_probability_models/index.md

```diff
@@ -0,0 +1,603 @@
+
+$$ x' = \text{sample}(P(X' \mid x)) $$ 
+
+and $x'$ will be our new particle in the set.
+
+### Observe
+Now the robot has new observations. We score every guess produced in the last steps by the new observation (give them weight) based on emission probability, which we have in HMMs, so, we know that how strong they are after new observation (similar to likelihood weighting). We give a weight to each particle by observing evidence $e$:
+
+$$ w(x) =  P(e \mid x)$$ 
+
+Be aware that we dont sample anything here and particles are fixed. Also note that the probabilities won't sum to one, as we are down-weighing almost every particle (some maybe very consistent with the evidence, and based on the approach of calculating the weight the can be one). 
+
+### Resample
+
+Working with weights can be frustrating for our little robot (!) and some can converge to zero after some iterations, so, based on how probable and strong our particles were, we generate a new set of particles. This work is done be sampling over the weights of the particles $N$ times (so the size of the particle set remain the same). The stronger a particle is, the more probable it is to be sampled and be in the new particle set. After this step we have a new set of particle which are distributed by the strength of the particles, which were calculated in observation step, that keep the frequency of the samples strong and valid. And we will go back to the "Elapse Time" step.
+
+### Recap
+That's all folks! First we have a set of particles. ‌Based on where they are each, we guess where they would be in the next step ahead in time. An observation is done by the robot. We score (weight) the guesses to know how probabil after the observation. And resample based on weights, to normalize particles. and we repeat this steps again and again till we converge.
+
+## Example
+
+| ![Particle Filtering ](./assets/particle-filter-example.jpg) | 
+|:--:| 
+| *An example of a full particle filtering process.* |
+
+## Pseudo Code
+
+```java
+function PARTICLE_FILTERING(e, N, dbn) returns a set of samples for the next time step
+	 inputs: e, the new incoming evidence
+	     N, the number of samples to be maintained
+	     dbn, a DBN with prior P(X0), transition model P(X1 | X0), sensor model P(E1 | X1)
+	 persistent: S, a vector of samples of size N, initially generated from P(X0)
+	 local variables: W, a vector of weights of size N
+
+	 for i = 1 to N do
+	   S[i] ← sample from P(X1 | X0 = S[i]) /* step 1 */
+	   W[i] ← P(e | X1 = S[i])       /* step 2 */
+	 S ← WEIGHTED_SAMPLE_WITH_REPLACEMENT(N, S, W)  /* step 3 */
+	 return S
+```
+
+## Useful links
```

Comment:
I think it's better to add a "useful links" section at the end of your notebook and mention all the links there, but that's your choice (just a suggestion)